src/main/resources/provision/aws/main.tf

```diff
@@ -0,0 +1,68 @@
+data "aws_eks_cluster" "cluster" {
+  name = module.eks.cluster_id
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = module.eks.cluster_id
+}
+
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.cluster.token
+  load_config_file       = false
+}
+
+data "aws_availability_zones" "available" {
+}
+
+locals {
+  cluster_name = "{{EKS_CLUSTER_NAME}}"
+}
+
+module "vpc" {
+  source  = "{{EKS_VPC_SOURCE}}"
+  version = "{{EKS_VPC_VERSION}}"
+
+  name                 = "{{EKS_VPC_NAME}}"
+  cidr                 = "172.16.0.0/16"
+  azs                  = data.aws_availability_zones.available.names
+  private_subnets      = ["172.16.1.0/24", "172.16.2.0/24", "172.16.3.0/24"]
+  public_subnets       = ["172.16.4.0/24", "172.16.5.0/24", "172.16.6.0/24"]
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+  enable_dns_hostnames = true
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                      = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"             = "1"
+  }
+}
+
+module "eks" {
+  source  = "{{EKS_SOURCE}}"
+  version = "{{EKS_VERSION}}"
+
+  cluster_name    = "${local.cluster_name}"
+  cluster_version = "{{EKS_CLUSTER_VERSION}}"
+  subnets         = module.vpc.private_subnets
+
+  vpc_id = module.vpc.vpc_id
+
+  node_groups = {
```

Comment:
All of that section has to be configurable.
Currently it is subjective configuration. I might want medium, not large, and another capacity.

Reply:
Keep it in terraform.AwsEksProvider class